File: students/students/src/MON.Models/StudentModels/Search/StudentSearchModel.cs
namespace MON.Models.StudentModels.Search
{
    public class StudentSearchModel
    {
        public string Pin { get; set; }
        public string FirstName { get; set; }
        public string MiddleName { get; set; }
        public string LastName { get; set; }
        public bool ExactMatch { get; set; }
        public string PublicEduNumber { get; set; }
        public string District { get; set; }
        public string Municipality { get; set; }
        public string School { get; set; }

        public int? PinType { get; set; }

        public bool? OnlyOwnInstitution { get; set; }


        public bool HasValue =>
            !string.IsNullOrWhiteSpace(Pin) ||
            !string.IsNullOrWhiteSpace(FirstName) ||
            !string.IsNullOrWhiteSpace(MiddleName) ||
            !string.IsNullOrWhiteSpace(LastName) ||
            !string.IsNullOrWhiteSpace(PublicEduNumber) ||
            !string.IsNullOrWhiteSpace(District) ||
            !string.IsNullOrWhiteSpace(Municipality) ||
            !string.IsNullOrWhiteSpace(School);
    }
}

File: students/students/src/MON.SPA/src/models/studentSearchModel.js
export class StudentSearchModel {
    constructor(obj = {}) {
      this.pin = obj.pin || '';
      this.firstName = obj.firstName || '';
      this.middleName = obj.middleName || '';
      this.lastName = obj.lastName || '';
      this.publicEduNumber = obj.publicEduNumber || '';
      this.district = obj.district || '';
      this.municipality = obj.municipality || '';
      this.school = obj.school || '';
      this.exactMatch = obj.exactMatch || true;
      this.filter = obj.filter || '';
      this.pageIndex = obj.pageIndex || 0;
      this.pageSize = obj.pageSize || 10;
      this.sortBy = obj.sortBy || '';
      this.onlyOwnInstitution = obj.onlyOwnInstitution || false;
    }
  }